symex-interface.el

```diff
@@ -0,0 +1,61 @@
+;;; symex-interface.el --- An evil way to edit Lisp symbolic expressions as trees  -*- lexical-binding: t; -*-
+
+;; URL: https://github.com/countvajhula/symex.el
+
+;; This program is "part of the world," in the sense described at
+;; https://drym.org.  From your perspective, this is no different than
+;; MIT or BSD or other such "liberal" licenses that you may be
+;; familiar with, that is to say, you are free to do whatever you like
+;; with this program.  It is much more than BSD or MIT, however, in
+;; that it isn't a license at all but an idea about the world and how
+;; economic systems could be set up so that everyone wins.  Learn more
+;; at drym.org.
+;;
+;; This work transcends traditional legal and economic systems, but
+;; for the purposes of any such systems within which you may need to
+;; operate:
+;;
+;; This is free and unencumbered software released into the public domain.
+;; The authors relinquish any copyright claims on this work.
+;;
+
+;;; Commentary:
+
+;; Miscellaneous Lisp editing-related features
+
+;;; Code:
+
+(defvar symex-interfaces '())
+
+(defvar symex-methods
+  (list :eval
+        :eval-definition
+        :eval-pretty
+        :eval-thunk
+        :eval-print
+        :describe-symbol
+        :repl
+        :run))
+
+(defun symex-interface-assert (interface)
+  "Assert that INTERFACE is a valid symex interface."
+  (mapc (lambda (k)
+          (cl-assert (plist-get interface k)
+                     (concat "symex interface: missing method: " (symbol-name k)) ))
```

Comment:
I would say we need not raise an error if these are not present, since some could be unspecified. Instead, we can provide a default implementation, as you suggested, for methods that are `nil`. For now, unless you have specific ideas on default implementation, I'm fine with just using `(lambda args (error "Not implemented"))`.

One possible implementation is for this function to return the modified interface, including the default methods replacing any that are `nil`, and in that case we'd probably want to rename this function something like `symex-interface-validate` or `symex-interface-finalize`.